385/Assets/Scripts/FireGrappleHook.cs

```diff
@@ -134,11 +134,16 @@ private void OnTriggerEnter2D(Collider2D collider)
     // be initiated).
     void OnCollisionEnter2D(Collision2D collision)
 	{
+		string colliderTag = collision.collider.tag;
 		// Don't check for collisions with the Player object
 		if (collision.collider.tag == "Player") 
 		{
-			Physics2D.IgnoreCollision (gameObject.GetComponent<Collider2D>(), collision.collider);
-			return;
+			Physics2D.IgnoreCollision (gameObject.GetComponent<Collider2D> (), collision.collider);
+		} 
+		// If the hook collides with the floor or a wall, reset
+		else if (colliderTag == "Ground" || colliderTag == "Floor & Wall") 

```

Comment:
can we define these as constant strings?

Reply:
good idea. I'll format the tag constant variables as TAG_NAME_HERE if that's fine

Reply:
see PlayerController as a reference